packages/insomnia-app/app/sync/vcs/index.ts

```diff
@@ -1163,32 +1172,98 @@ export default class VCS {
     return project.teams;
   }
 
-  async _queryCreateProject(workspaceId: string, workspaceName: string) {
-    const { publicKey } = this._assertSession();
+  async _queryTeamMemberKeys(
+    teamId: string,
+  ): Promise<{
+    memberKeys: {
+      accountId: string;
+      publicKey: string;
+    }[];
+  }> {
+    const { teamMemberKeys } = await this._runGraphQL(
+      `
+        query ($teamId: ID!) {
+          teamMemberKeys(teamId: $teamId) {
+            memberKeys {
+              accountId
+              publicKey
+            }
+          }
+        }
+      `,
+      {
+        teamId: teamId,
+      },
+      'teamMemberKeys',
+    );
+    return teamMemberKeys;
+  }
 
+  async _queryCreateProject(
+    workspaceId: string,
+    workspaceName: string,
+    teamId?: string,
+    teamPublicKeys?: {
+      accountId: string;
+      publicKey: string;
+    }[],
+  ) {
     // Generate symmetric key for ResourceGroup
     const symmetricKey = await crypt.generateAES256Key();
     const symmetricKeyStr = JSON.stringify(symmetricKey);
-    // Encrypt the symmetric key with Account public key
-    const encSymmetricKey = crypt.encryptRSAWithJWK(publicKey, symmetricKeyStr);
+
+    const teamKeys: any[] = [];
+    let encSymmetricKey: string | undefined;
+
+    if (teamId && teamPublicKeys) {
```

Comment:
Should this check be the following?

```suggestion
    if (teamId && teamPublicKeys?.length > 0) {
```

What should happen if teamPublicKeys is an empty array - is it safe to assume it never will be an empty array?

Reply:
The API will just reject it as `teamPublicKeys` must not be empty if teamId is set, but I rewrote this check to be a bit more defensive, and so that it's more obvious as well.

Reply:
Awesome 🙌🏽